samples/OpenIDConnectSample/Startup.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.AspNet.Security.OpenIdConnect;
+using Microsoft.AspNet.Http.Security;
+using Microsoft.AspNet.Security;
+using Microsoft.AspNet.Security.Cookies;
+
+namespace OpenIDConnectSample
+{
+    public class Startup
+    {
+        public void Configure(IApplicationBuilder app)
+        {
+            app.UseServices(services =>
+            {
+                services.AddDataProtection();
+                services.Configure<ExternalAuthenticationOptions>(options =>
+                {
+                    options.SignInAsAuthenticationType = OpenIdConnectAuthenticationDefaults.AuthenticationType;
+                });
+
+            });
+
+            app.UseCookieAuthentication(options =>
+            {
+                options.AuthenticationType = OpenIdConnectAuthenticationDefaults.AuthenticationType;
```

Comment:
And the sample is working as expected?

If it is, that's really sad: IMO, having multiple authentication handlers configured with the same `AuthenticationType` is really bad and should trigger an exception somewhere in `Microsoft.AspNet.Security` as it can't be a valid scenario (here, the `AuthenticationType` is shared with the OIDC middleware).

IIRC, that was not the case with Katana 3, but an `InvalidOperationException` thrown by LINQ was thrown when you called `AuthenticateAsync` with an `AuthenticationType` shared my multiple handlers.

If @Tratcher agrees with the general concept, I suggest opening a new ticket to track that.